scripts/Tools/testreporter.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python
+
+"""
+
+Simple script to populate CESM test database with test results. 
+
+"""
+
+from standard_script_setup import *
+
+from CIME.XML.env_build             import EnvBuild
+from CIME.XML.env_case              import EnvCase
+from CIME.XML.env_test              import EnvTest
+from CIME.XML.test_reporter         import TestReporter
+from CIME.utils                     import expect
+
+
+import glob
+
+
+
+
+
+# Parse command line options
+
+parser = argparse.ArgumentParser(description='Arguements for testreporter')
+parser.add_argument("--tagname", 
+                        help="Name of the tag being tested.")
+parser.add_argument("--testid", 
+                        help="Test id, ie c2_0_a6g,c2_0_b6g.")
+parser.add_argument("--testroot", 
+                        help="Root directory for tests to populate the database.")
+parser.add_argument("--testtype", 
+                        help="Type of test, prealpha or prebeta.")
+parser.add_argument("--dryrun",action="store_true", 
+                        help="Do a dry run, database will not be populated.")
+parser.add_argument("--dumpxml",action="store_true", 
+                        help="Dump XML test results to sceen.")
+args = parser.parse_args()
+
+# Fill in values needed from xml files (COMPILER, MPILIB, MACH, BASELINE_NAME_CMP.
+
+os.chdir(args.testroot)
+
+#
+# Retrieve compiler name and mpi library
+#
+xml_file=glob.glob("*"+args.testid+"/env_build.xml")
+expect(len(xml_file) > 0, "Tests not found.  It's possible your testid, %s is wrong." %args.testid )
+print len(xml_file)
+envxml=(EnvBuild(".",infile=xml_file[0]))
+compiler=envxml.get_value("COMPILER")
+mpilib=envxml.get_value("MPILIB")
+
+#
+# Retrieve machine name
+#
+xml_file=glob.glob("*"+args.testid+"/env_case.xml")
+envxml=(EnvCase(".",infile=xml_file[0]))
+machine=envxml.get_value("MACH")
+
+#
+# Retrieve baseline tag to compare to 
+#
+xml_file=glob.glob("*"+args.testid+"/env_test.xml")
+envxml=(EnvTest(".",infile=xml_file[0]))
+baseline = envxml.get_value("BASELINE_NAME_CMP")
+
+#
+# Create XML header
+#
+
+testxml=TestReporter()
+testrecord=None
+testxml.setup_header(args.tagname,machine,compiler,mpilib,args.testroot,args.testtype,baseline)
+
+#
+# Create lists on tests based on the testid in the testroot directory.
+#
+test_names=glob.glob("*"+args.testid)
+#
+# Loop over all tests and parse the test results
+#
+test_status={}
+for test_name in test_names:  
```

Comment:
The 3 "tests" above are not tests.  They're pulling out information needed to populated the test database.  It is possible for a test to fail without creating any env*.xml files.  So, testing env*xml files would prove nothing.  However, a TestStatus file  is always created.  So, I'll replace the check for cs.status+test_id, with a test for the existence of test_name/TestStatus.  That way I'll account for other files in the testroot directory which end with the testid string.